From 98d16f1b5ee12a54fdb5a644db02f8011528ec91 Mon Sep 17 00:00:00 2001
Message-ID: <98d16f1b5ee12a54fdb5a644db02f8011528ec91.1747779547.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1747779547.git.sharlatanus@gmail.com>
References: <cover.1747779547.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 11 May 2025 20:34:14 +0100
Subject: [PATCH 05/31] gnu: Add python-numpy-stl.

* gnu/packages/python-xyz.scm (python-numpy-stl): New variable.

Change-Id: Iec45ccf1a2c83dc608769971fe0d2e5b20b20343
---
 gnu/packages/python-xyz.scm | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index 85ed810534..6fa9482ae6 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -10096,6 +10096,33 @@ (define-public python-numpy-documentation
     (description "This package provides the complete NumPy documentation in
 the Texinfo, HTML, and PDF formats.")))
 
+(define-public python-numpy-stl
+  (package
+    (name "python-numpy-stl")
+    (version "3.2.0")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "numpy_stl" version))
+       (sha256
+        (base32 "0kgxhghl9j5caydb2a9g2n44zvffd94gb6abday0manxkkvw682s"))))
+    (build-system pyproject-build-system)
+    (native-inputs
+     (list python-cython
+           python-pytest
+           python-setuptools
+           python-wheel))
+    (propagated-inputs
+     (list python-numpy
+           python-utils))
+    (home-page "https://github.com/WoLpH/numpy-stl/")
+    (synopsis "Read, write and modify binary/ASCII STL files library")
+    (description
+     "Library to make reading, writing and modifying both binary and ascii STL
+files, file format native to the stereolithography CAD software created by 3D
+Systems, easy.")
+    (license license:bsd-3)))
+
 (define-public python-npx
   (package
     (name "python-npx")
-- 
2.49.0

